stdlib/Distributed/src/cluster.jl

```diff
@@ -658,7 +671,7 @@ function create_worker(manager, wconfig)
         delete!(PGRP.refs, ntfy_oid)
     end
 
-    return w.id
+    return (istaskdone(procmsg_task) || (first(rr_ntfy_join.c.data) !== :OK)) ? 0 : w.id
```

Comment:
Replace `first(rr_ntfy_join.c.data)` with `fetch(rr_ntfy_join.c)`. Avoids using internals of the `Channel` implementation.

Also the `istaskdone(procmsg_task)` check can be removed since we will come here only after `wait(rr_ntfy_join)` has been triggered.

Reply:
Done.